src/metal.cs

```diff
@@ -2892,8 +3173,154 @@ interface MTLTileRenderPipelineDescriptor : NSCopying {
 		[Export ("tileBuffers")]
 		MTLPipelineBufferDescriptorArray TileBuffers { get; }
 
+		[iOS (12,0)]
+		[Export ("maxTotalThreadsPerThreadgroup")]
+		nuint MaxTotalThreadsPerThreadgroup { get; set; }
+
 		[Export ("reset")]
 		void Reset ();
 	}
+
+	interface IMTLEvent {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[Protocol]
+	interface MTLEvent {
+		[Abstract]
+		[NullAllowed, Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DesignatedDefaultCtor]
+	interface MTLSharedEventListener {
+		[Export ("initWithDispatchQueue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (DispatchQueue dispatchQueue);
+
+		[Export ("dispatchQueue")]
+		DispatchQueue DispatchQueue { get; }
+	}
+
+	delegate void MTLSharedEventNotificationBlock (IMTLSharedEvent arg0, ulong arg1);
+
+	interface IMTLSharedEvent {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[Protocol]
+	interface MTLSharedEvent : MTLEvent {
+		[Abstract]
+		[Export ("notifyListener:atValue:block:")]
+		void NotifyListener (MTLSharedEventListener listener, ulong atValue, MTLSharedEventNotificationBlock block);
+
+		[Abstract]
+		[Export ("newSharedEventHandle")]
+		MTLSharedEventHandle CreateSharedEventHandle ();
+
+		[Abstract]
+		[Export ("signaledValue")]
+		ulong SignaledValue { get; set; }
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MTLSharedEventHandle : NSSecureCoding {
+		[NullAllowed, Export ("label")]
+		string Label { get; }
+	}
+
+	interface IMTLIndirectRenderCommand {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[Protocol]
+	interface MTLIndirectRenderCommand {
+		[NoiOS, NoTV, Mac (10,14, onlyOn64: true)]
+		[Abstract]
+		[Export ("setRenderPipelineState:")]
+		void SetRenderPipelineState (IMTLRenderPipelineState pipelineState);
```

Comment:
Again, we could remove the RenderPipelineState since it can be deduced from the first arg.